File: day11/hex_ed.py
#!/usr/bin/env python 3
from sys import argv

filename = argv[1]
with open(filename) as f:
    for line in f:
        line = line.rstrip("\n")
        moves = line.split(",")
        row = 0
        col = 0
        far = 0
        for move in moves:
            if move == "n":
                row += 2
            elif move == "ne":
                row += 1
                col += 1
            elif move == "se":
                row -= 1
                col += 1
            elif move == "s":
                row -= 2
            elif move == "sw":
                row -= 1
                col -= 1
            elif move == "nw":
                row += 1
                col -= 1

            d = int((abs(row) + abs(col)) / 2)
            far = max(far, d)

        d = int((abs(row) + abs(col)) / 2)
        print("result1:", d)
        print("result2:", far)

File: day11/hex_ed.pl
#!/usr/bin/env perl
use strict;
use warnings;
use feature qw(:5.24);
use experimental qw(signatures);

while (<>) {
    chomp;
    my @moves = split /,/;
    my ($row, $col) = (0, 0);
    my $far = 0;
    for my $move (@moves) {
	if ($move eq "n") {
	    $row += 2;
	} elsif ($move eq "ne") {
	    $row++;
	    $col++;
	} elsif ($move eq "se") {
	    $row--;
	    $col++;
	} elsif ($move eq "s") {
	    $row -= 2;
	} elsif ($move eq "sw") {
	    $row--;
	    $col--;
	} elsif ($move eq "nw") {
	    $row++;
	    $col--;
	}
	my $d = (abs($row) + abs($col)) / 2;
	$far = $d if $d > $far;
    }
    my $d = (abs($row) + abs($col)) / 2;
    say "result1: $d";
    say "result2: $far";
}
